prototype_source/pt2e_quant_ptq_static.rst

```diff
@@ -0,0 +1,399 @@
+
+
+4. Import the Backend Specific Quantizer and Configure how to Quantize the Model
+--------------------------------------------------------------------------------
+
+.. code:: python
+
+  from torch.ao.quantization.xnnpack_quantizer import (
+    XNNPackQuantizer,
+    get_symmetric_quantization_config,
+  )
+  quantizer = XNNPackQuantizer()
+  quantizer.set_globa(get_symmetric_quantization_config())
+
+`Quantizer` is backend specific, and each `Quantizer` will provide their own way to allow users to configure their model. Just as an example, here is the different configuration APIs supported by XNNPackQuantizer:
+
+.. code:: python
+  quantizer.set_global(qconfig_opt)  # qconfig_opt is an optional qconfig, either a valid qconfig or None
+      .set_object_type(torch.nn.Conv2d, qconfig_opt) # can be a module type
+      .set_object_type(torch.nn.functional.linear, qconfig_opt) # or torch functional op      
+      .set_module_name("foo.bar", qconfig_opt)
+
+5. Prepare the Model for Post Training Static Quantization
+----------------------------------------------------------
+
+`prepare_pt2e` folds BatchNorm modules into previous Conv2d modules, and insert observers
+in appropriate places in the model.
+
+.. code:: python
+
+    prepared_model = prepare_pt2e(model_to_quantize, quantizer)
+    print(prepared_model.graph)
+
+6. Calibration
+--------------
+Calibration function is run after the observers are inserted in the model.
+The purpose for calibration is to run through some sample examples that is representative of the workload
+(for example a sample of the training data set) so that the observers in the model are able to observe
+the statistics of the Tensors and we can later use this information to calculate quantization parameters.
+
+.. code:: python
+
+    def calibrate(model, data_loader):
+        model.eval()
+        with torch.no_grad():
+            for image, target in data_loader:
+                model(image)
+    calibrate(prepared_model, data_loader_test)  # run calibration on sample data
+
+7. Convert the Model to a Quantized Model
+-----------------------------------------
+``convert_pt2e`` takes a calibrated model and produces a quantized model.
+
+.. code:: python
+
+    quantized_model = convert_pt2e(prepared_model)
+    print(quantized_model)
+
+Note: the model produced here also had some improvement upon the previous `representations <https://github.com/pytorch/rfcs/blob/master/RFC-0019-Extending-PyTorch-Quantization-to-Custom-Backends.md>` in fx graph mode quantizaiton, previously all quantized operators are represented as `dequantize -> fp32_op -> qauntize`, in the new flow, we choose to represent some of the operators with integer computation so that it's closer to the computation happens in hardwares. For more details, please see: `Quantized Model Representation <https://docs.google.com/document/d/17h-OEtD4o_hoVuPqUFsdm5uo7psiNMY8ThN03F9ZZwg/edit>` (TODO: make this an API doc/issue).
```

Comment:
```suggestion
.. note::  The model produced here also had some improvement upon the previous `representations <https://github.com/pytorch/rfcs/blob/master/RFC-0019-Extending-PyTorch-Quantization-to-Custom-Backends.md>`_ in the FX graph mode quantization, previously all quantized operators are represented as ``dequantize -> fp32_op -> qauntize``, in the new flow, we choose to represent some of the operators with integer computation so that it's closer to the computation happens in hardwares. For more details, please see: `Quantized Model Representation <https://docs.google.com/document/d/17h-OEtD4o_hoVuPqUFsdm5uo7psiNMY8ThN03F9ZZwg/edit>`_.
```